### PR TITLE
Alert Management Logic

### PR DESCRIPTION
**Pull request by eyingxuan**

### CODE REVIEW

File: frontend/alert/src/components/ManageAlertLogic.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            });
+            sections = Array.from(sections);
+
+            const promises = [];
+            sections.forEach((section) => {
+                promises.push(getStatusPromise(section));
+            });
+
+            Promise.all(promises)
+                .then(vals => Promise.all(vals.map(val => val.json())))
```
Comment (davish) [inline frontend/alert/src/components/ManageAlertLogic.js:37]: Could this request happen inside of the `AlertItem` component? Since it's tied directly to that component. Might make this code cleaner to read / shorter.

**Comment by eyingxuan** [comment]
Branch is ready for review! Features added include 
- Fetching registrations data from backend
- Action buttons to resubscribe/cancel for individual alerts
- Batch select/actions to resubscribe/cancel/delete
- Filter alerts by section id

File: frontend/alert/src/App.js [hunk 2]
```diff
@@ -3,12 +3,9 @@ import "./App.css";
 import styled from "styled-components";
 import PropTypes from "prop-types";
 import Logo from "./assets/PCA_logo.svg";
-
+import ManageAlertWrapper from "./components/managealert/index";
```
Comment (davish) [inline frontend/alert/src/App.js:6]:
```suggestion
import ManageAlertWrapper from "./components/managealert";
```
index.js files are the default for the directory, so no need to add `index`! I'd also recommend renaming the directory to `ManageAlerts` to match casing for components.

File: frontend/alert/src/components/managealert/AlertItem.js [hunk 3]
```diff
@@ -60,10 +60,21 @@ export const AlertItem = ({
     return (
         <>
             <GridItem column="1" row={rownum} border halign valign>
-                <input type="checkbox" />
+                <input type="checkbox" checked={checked} onChange={toggleAlert} />
             </GridItem>
             <GridItem column="2" row={rownum} border valign>
-                <P size="0.7rem">{date}</P>
+                {
+                    date
```
Comment (davish) [inline frontend/alert/src/components/managealert/AlertItem.js:67]: what date does this represent? from the context I'd guess it has to do with the last time an alert was sent for this registration. Might be better in that case to rename this prop `alertLastSent` or something along that line.

File: frontend/alert/src/components/managealert/AlertItem.js [hunk 4]
```diff
@@ -21,7 +21,7 @@ const StatusGridItem = styled(GridItem)`
 
 // Component for an alert entry (renders as a row in CSS grid)
 export const AlertItem = ({
-    date, course, status, repeat, actions, rownum,
+    date, course, status, repeat, actions, rownum, checked, toggleAlert, actionHandler,
```
Comment (davish) [inline frontend/alert/src/components/managealert/AlertItem.js:24]: would rename `actionHandler` to something more specific -- what action is this for? When is it invoked?

File: frontend/alert/src/components/managealert/AlertSearch.js [hunk 5]
```diff
@@ -25,11 +26,16 @@ const SearchBarFlex = styled(Flex)`
 
 // Component for search filter
 // in alert management
-export const AlertSearch = () => (
+export const AlertSearch = ({ value, onChange }) => (
     <SearchFlex valign>
         <SearchBarFlex valign margin="0.2rem">
             <Img src={Search} alt="" width="0.6rem" height="0.6rem" />
-            <SearchInput type="search" placeholder="Search" />
+            <SearchInput type="search" placeholder="Search" value={value} onChange={onChange} />
```
Comment (davish) [inline frontend/alert/src/components/managealert/AlertSearch.js:33]: This can be a TODO for another PR to use the `AutoComplete` component from @dlike230.

File: frontend/alert/src/components/managealert/index.js [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+const fetchAlerts = () => (
+    fetch("/api/alert/api/registrations")
+        .then(res => res.json())
+);
+
+const processAlerts = (setAlerts) => {
+    const fetchPromise = () => (
+        fetchAlerts()
+            .then(res => (
+                res.map((section) => {
```
Comment (davish) [inline frontend/alert/src/components/managealert/index.js:16]: `section` might be misleading here. Would rename to `registration`

File: frontend/alert/src/components/managealert/index.js [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+                        }
+                    } else {
+                        repeat = AlertRepeat.Inactive;
+                    }
+
+                    return {
+                        id: section.id,
+                        original_created_at: section.original_created_at,
+                        section: section.section,
+                        date: datetime,
```
Comment (davish) [inline frontend/alert/src/components/managealert/index.js:49]: would rename to `lastNotified`

File: frontend/alert/src/components/managealert/index.js [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+                        } else {
+                            repeat = AlertRepeat.Once;
+                        }
+                    } else {
+                        repeat = AlertRepeat.Inactive;
+                    }
+
+                    return {
+                        id: section.id,
+                        original_created_at: section.original_created_at,
```
Comment (davish) [inline frontend/alert/src/components/managealert/index.js:47]: this would be a good spot to transition from python snake_case to JS camelCase.
```suggestion
                        originalCreatedAt: section.original_created_at,
```

Comment (davish) [reply frontend/alert/src/components/managealert/index.js:47]: I'm guessing this would require some find-and-replace in other places

File: frontend/alert/src/components/managealert/index.js [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+            Accept: "application/json",
+            "Content-Type": "application/json",
+            "X-CSRFToken": getCsrf(),
+        },
+        body: JSON.stringify(body),
+    });
+};
+
+
+const actionHandler = callback => (id, actionenum) => {
```
Comment (davish) [inline frontend/alert/src/components/managealert/index.js:110]: This is basically a mini bit of redux here -- a descriptive comment would go a ways towards illuminating that for someone adding more functionality down the line.

File: frontend/alert/src/components/managealert/index.js [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+    useEffect(() => {
+        processAlerts(setAlerts);
+    }, []);
+
+    return (
+        <>
+            <ManageAlertHeader />
+            <ManageAlert
+                setFilter={setFilter}
+                numSel={numSelected}
```
Comment (davish) [inline frontend/alert/src/components/managealert/index.js:172]: If this is just derived from `alertSel`, it might make sense to just have it computed in the `ManageAlert` component.

File: frontend/alert/src/App.js (hunk 2 above)
Comment (eyingxuan) [reply frontend/alert/src/App.js:6]: oh wow! i did not know they default to index.js - learned something new :)

Comment (eyingxuan) [reply frontend/alert/src/App.js:6]: We actually get a linter error for having a directory name with uppercases - seems like it's due to the fact that some file systems are case insensitive, and that screws with the code base.

Comment (eyingxuan) [reply frontend/alert/src/App.js:6]: jk, this seems to be there since the beginning before I even moved directories. It seems to have not come from ESLint, but my editor. I'll investigate

Comment (eyingxuan) [reply frontend/alert/src/App.js:6]: Never mind - the directory was the issue, after resetting to head my projectile cache still used the upper case directory which is why my linter was complaining. I'm gonna use lower case directory names then.

File: frontend/alert/src/components/managealert/AlertSearch.js (hunk 5 above)
Comment (eyingxuan) [reply frontend/alert/src/components/managealert/AlertSearch.js:33]: This might be the wrong place for the `AutoComplete` component. I think you might be thinking of `ManageAlertHeader` which is where users actually search for alerts. This search merely lets them filter their own registrations
